pkg/collector/metric/node_metric.go

```diff
@@ -85,11 +84,29 @@ func getCPUArchitecture() (string, error) {
 		klog.V(2).Infof("cpu arch override: %v\n", cpuArchOverride)
 		return cpuArchOverride, nil
 	}
-	output, err := exec.Command("archspec", "cpu").Output()
+
+	// use cpuid to get CPUArchitecture
+	grep := exec.Command("grep", "uarch")
+	output := exec.Command("cpuid", "-1")
```

Comment:
per `cpuid` [manpage](https://man7.org/linux/man-pages/man4/cpuid.4.html), it is a x86 tool.  Can you refactor the code to externalize the `getCPUArchitecture` call in separate files and conditionally compile it to use `cpuid` on x86 and `archspec` on arm64?

Reply:
Updated. Move all unrelated node_metric func to utils.go and distinguish x86/arm64. (if you think it's better to create a new file for cpu architecture, I can edit it again).